src/Bridge/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -130,9 +133,42 @@ protected function filterProperty(string $property, $value, QueryBuilder $queryB
             $wrapCase = $this->createWrapCase($caseSensitive);
             $valueParameter = $queryNameGenerator->generateParameterName($field);
 
-            $queryBuilder
-                ->andWhere(sprintf($wrapCase('%s.%s').' IN (:%s)', $alias, $field, $valueParameter))
-                ->setParameter($valueParameter, $caseSensitive ? $values : array_map('strtolower', $values));
+            /*
+             * If field type is string/float, Doctrine does not call convertToDatabaseValueSQL() because it
+             * does not know it needs conversion.
+             * This would lead to incorrect values for Ramsey\Uuid\Doctrine\UuidBinaryType for example.
+             * The only fix is to provide field type to doctrine ...
+             * BUT it is not possible with setParameter(), as third arg can only be
+             * \Doctrine\DBAL\Connection::PARAM_STR_ARRAY
+             *
+             * The only way to do this will be to rewrite the IN() statement to multiple values,
+             * each provided as a single setParameter()
+             */
+
+            // let's do a very simple query if we don't need the complex one (99% of time)
+            if (\in_array($metadata->getTypeOfField($field), ['string', 'integer'], true)) {
+                $queryBuilder
+                    ->andWhere(sprintf($wrapCase('%s.%s').' IN (:%s)', $alias, $field, $valueParameter))
+                    ->setParameter($valueParameter, $caseSensitive ? $values : array_map('strtolower', $values));
+            } else {
+                $inQuery = [];
+                $args = [];
+                $type = $metadata->getTypeOfField($field);
+
+                for ($i = 0; $i < \count($values); ++$i) {
+                    $inQuery[] = ':filterproperty'.$i;
+                    $args[] = new Parameter('filterproperty'.$i, $caseSensitive ? $values[$i] : strtolower($values[$i]), $type);
+                }
+
+                // setParameters reset current parameters set, so we need to merge with getParameters()
+                $parameters = new ArrayCollection($args);
```

Comment:
Why don't you create an `ArrayCollection` directly instead of creating an intermediate array?
Actually, you could even just use the `ArrayCollection` returned by `$queryBuilder->getParameters()`.